### DISCUSSION TITLE
Return effected rows with onConflictAction set to ignore

### DISCUSSION THREAD

**Discussion by ghostoy** [mikro-orm/mikro-orm#6182] [Ideas]
**Is your feature request related to a problem? Please describe.**
I want to know if the row is inserted or not when using `upsert` with `onConflictAction` set to `ignore`. Currently there is no way to get such results.

**Describe the solution you'd like**
Maybe MikroORM could add an option to `upsert` to return effected rows with the entity, such as:

```typescript
const { effectedRows, entity } = await em.upsert(MyEntity, { foo: 'bar' }, { onConflictAction: 'ignore', returnEffectedRows: true });
// while keep the method signature when `returnEffectedRows` is `false` as below:
const entity = await em.upsert(MyEntity, { foo: 'bar' }, { onConflictAction: 'ignore' });
```

**Describe alternatives you've considered**
A quick thought to workaround this is to simply `select-and-insert` in a transaction as such:

```typescript
await em.transactional(async em => {
  const entity= await em.findOne(MyEntity, { foo: 'bar' }, { lockMode: LockMode.PESSIMISTIC_WRITE }); // lock for update
  if (!entity) {
    const entity = em.create({ foo: 'bar' });
    await em.persistAndFlush(entity);
    return entity;
  }
  return entity;
});
```
This doesn't work if record does not exist. If multiple concurrent requests running in parallel, they may result in inserting multiple identical rows or throw conflict sql error.

**Additional context**
No.

**Comment by B4nan** [comment]
You can use QB for this already. Note that with the ignore option you wont get any results back from the database if the row exists.

```ts
const res = await em.createQueryBuilder(MyEntity).insert({ foo: 'bar' }).onConflict().ignore().execute();

if (res.affectedRows) {
  // created
  const entity = em.map(MyEntity, res.rows[0]);
} else {
 // already exists
}
```

I don't like the idea of changing return type based on some options, that's a code smell to me.
